### PR TITLE
listen to oplog when a transition to cold workflow is disabled

### PR DESCRIPTION
**Pull request by welansari**
This is needed to keep restore operations on cold objects working

Issue: BB-457

### CODE REVIEW

File: extensions/lifecycle/LifecycleOplogPopulatorUtils.js [hunk 1]
```diff
@@ -1,27 +1,59 @@
...
     /**
      * Get extension specific MongoDB filter
      * to get buckets that have the lifecycle
      * extension enabled
      * @returns {Object} MongoDB filter
      */
      static getExtensionMongoDBFilter() {
-        // getting buckets with at least one lifecycle
-        // rule enabled
+        const coldLocations = Object.keys(locations).filter(loc => locations[loc].isCold);
```
Comment (KillianG) [inline extensions/lifecycle/LifecycleOplogPopulatorUtils.js:13]: Why not just `locations.filter(loc => loc.isCold);` ?

Comment (welansari) [reply extensions/lifecycle/LifecycleOplogPopulatorUtils.js:13]: locations is an object not an array

File: extensions/lifecycle/LifecycleOplogPopulatorUtils.js [hunk 2]
```diff
@@ -1,27 +1,59 @@
...
             'value.lifecycleConfiguration.rules': {
                 $elemMatch: {
-                    ruleStatus: 'Enabled',
-                    actions: {
-                        $elemMatch: {
-                            $or: [
-                                { actionName: 'Transition' },
-                                { actionName: 'NoncurrentVersionTransition' },
-                            ],
+                    $or: [
```
Comment (francoisferrand) [inline extensions/lifecycle/LifecycleOplogPopulatorUtils.js]: not sure what is the most "efficient" (for mongo), or simple: could/shoud we check the following instead?

```
$elemMatch: {
   $and: [
       actions: {
          $elemMatch: {
              actionName: {
                  $in: ['Transition', 'NoncurrentVersionTransition'],
              },
          },
       },
      $or: [
         { ruleStatus: 'Enabled' },
         { actions: {
		transition: {
                    $elemMatch: { storageClass: { $in: coldLocations }},
                 },
         }},
         { actions: {
		nonCurrentVersionTransition: {
                    $elemMatch: { storageClass: { $in: coldLocations }},
                 },
         }},
      ]
   ]
}
```

Comment (francoisferrand) [reply extensions/lifecycle/LifecycleOplogPopulatorUtils.js]: (seems conceptually simple, with less redundancy ; but not sure if it performs better or is more readable when finalized...)

**Review by KillianG — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**